src/Message/SmsMessageResult.php

```diff
@@ -13,147 +13,128 @@
    *
-   * @var integer
-   */
-  public $creditBalance;
-
-  /**
-   * The message delivery reports keyed by recipient number.
+   * This number is in the SMS gateway's chosen denomination.
    *
-   * @var \Drupal\sms\Message\SmsDeliveryReportInterface[]
+   * @var float|NULL
    */
-  public $reports;
+  public $creditBalance = NULL;
 
   /**
-   * The messages associated with this result.
+   * The credits consumed to process this message, or NULL if unknown.
    *
-   * @var \Drupal\sms\Message\SmsMessageInterface[]
+   * This number is in the SMS gateway's chosen denomination.
+   *
+   * @var float|NULL
    */
-  public $messages = [];
+  public $creditsUsed = NULL;
 
   /**
-   * Create a new message result based on data supplied in the array.
-   *
-   * @param array $data
-   *   Information to be used to instantiate the SmsMessageResult.
-   */
-  public function __construct($data) {
-    $data += $this->defaultData();
-    $this->status = $data['status'];
-    $this->creditBalance = $data['credit_balance'];
-    $this->creditsUsed = $data['credits_used'];
-    $this->errorMessage = $data['error_message'];
-    $this->reports = $data['reports'];
+   * {@inheritdoc}
+   */
+  public function getStatus() {
+    return $this->status;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getStatus() {
-    return $this->status;
+  public function setStatus($status) {
+    $this->status = $status;
+    return $this;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getReports() {
-    return $this->reports;
+  public function getStatusMessage() {
+    return $this->statusMessage;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getBalance() {
-    return $this->creditBalance;
+  public function setStatusMessage($message) {
+    $this->statusMessage = $message;
+    return $this;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getCreditsUsed() {
-    return $this->creditsUsed;
+  public function getReport($recipient) {
+    foreach ($this->reports as $report) {
+      if ($report->getRecipient() == $recipient) {
+        return $report;
+      }
+    }
+    return NULL;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function toArray() {
-    return array(
-      'status' => $this->status,
-      'error_message' => $this->errorMessage,
-      'credits_used' => $this->creditsUsed,
-      'credit_balance' => $this->creditBalance,
-      'reports' => $this->reports,
-    );
+  public function getReports() {
+    return $this->reports;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getErrorMessage() {
-    return $this->errorMessage;
+  public function setReports(array $reports) {
+    $this->reports = $reports;
+    return $this;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getReport($recipient) {
-    if (isset($this->reports[$recipient])) {
-      return $this->reports[$recipient];
-    }
-    else {
-      return null;
-    }
+  public function getCreditsBalance() {
```

Comment:
I think either works. Plural makes things line up nicely since `getCreditsUsed` has the same prefix.